### ISSUE TITLE
High auc_score and very low precision_at_k, recall_at_k, reciprocal_rank.

### ISSUE THREAD

**Issue by MaxBalashov** [lyst/lightfm#205]
My input:
interactions - sparse matrix <666x18962>,
user_features - sparse matrix <666x514>,
item_features - sparse matrix <18962x259>.

I use cross-validation for estimate all metrics on the train and test sets.
And I get high auc_score (0.9), but very low other metrics (0.05-0.15).
Also I used different loss functions, but in general this situation is stable. 
I understand that every metric has unique formula.

```
auc_score
0.7629 (logistic)
0.8022 (bpr)
0.9204 (warp)
0.9297 (warp-kos)
####################
precision_at_k
0.0406 (logistic)
0.0448 (bpr)
0.0449 (warp)
0.0474 (warp-kos)
####################
recall_at_k
0.09 (logistic)
0.0893 (bpr)
0.0869 (warp)
0.081 (warp-kos)
####################
reciprocal_rank
0.1148 (logistic)
0.1208 (bpr)
0.1249 (warp)
0.1367 (warp-kos)
####################
```

Does it mean (high auc, very low others ) that my recommender predict is bad, and I cannot use it?

**Comment by maciejkula** [comment]
The three metrics you mention measure different things, so it's perfectly natural for a successful model to have different AUC, precision, MRR, and recall scores. I would look at their Wikipedia definitions to get a feel for how to interpret them; for what it's worth, your models' scores look fine!
